src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -96,18 +96,49 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
         return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
             .then(state => new Promise<KlarnaLoadResponse>((resolve, reject) => {
                 const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                const shippingAddress = state.shippingAddress.getShippingAddress();
+                const billingAddress = state.billingAddress.getBillingAddress();
 
                 if (!paymentMethod) {
                     throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
                 }
 
+                if (!billingAddress) {
+                    throw new MissingDataError(MissingDataErrorType.MissingBillingAddress);
+                }
+
+                if (!shippingAddress) {
+                    throw new MissingDataError(MissingDataErrorType.MissingShippingAddress);
+                }
+
                 if (!this._klarnaCredit || !paymentMethod.clientToken) {
                     throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
                 }
 
                 this._klarnaCredit.init({ client_token: paymentMethod.clientToken });
 
-                this._klarnaCredit.load({ container }, response => {
+                this._klarnaCredit.load({ container }, {
+                    billing_address: {
```

Comment:
You can take all this object into a separate method to reduce lines in single method i.e. `billing_address: mapKlarnaBillingAddress(billingAddress) ` also this comment apply to shipping_address